scripts/download-realm.js

```diff
@@ -25,9 +25,36 @@ const child_process = require('child_process');
 const fetch = require('node-fetch');
 const ini = require('ini');
 const decompress = require('decompress');
+const crypto = require('crypto');
 
 const LOCKFILE_NAME = 'download-realm.lock';
 
+const TEMP_DIR_SUFFIX = generateRandomString();
+
+function generateRandomString() {
+    return crypto.randomBytes(20).toString('hex');
+}
+
+function ensureDirExists(dir) {
+    if (!fs.existsSync(dir)) {
+        fs.mkdir(dir);
```

Comment:
Shouldn't this be `mkdirSync` otherwise it might not be ready exiting this method?